### ISSUE TITLE
[kind local registry]: http://localhost:5000 is not reachable 

### ISSUE THREAD

**Issue by iamtodor** [kubernetes-sigs/kind#2665]
I run local registry script:

<details>

```
#!/bin/sh
set -o errexit

# create registry container unless it already exists
reg_name='kind-registry'
reg_port='5001'
if [ "$(docker inspect -f '{{.State.Running}}' "${reg_name}" 2>/dev/null || true)" != 'true' ]; then
  docker run \
    -d --restart=always -p "127.0.0.1:${reg_port}:5000" --name "${reg_name}" \
    registry:2
fi

# create a cluster with the local registry enabled in containerd
cat <<EOF | kind create cluster \
--image kindest/node:v1.19.11 \
--config=-
kind: Cluster
apiVersion: kind.x-k8s.io/v1alpha4
containerdConfigPatches:
- |-
  [plugins."io.containerd.grpc.v1.cri".registry.mirrors."localhost:${reg_port}"]
    endpoint = ["http://${reg_name}:5000"]
nodes:
  - role: control-plane
    extraMounts:
      - hostPath: /projects/airflow_dags/
        containerPath: /data_dags
EOF

# connect the registry to the cluster network if not already connected
if [ "$(docker inspect -f='{{json .NetworkSettings.Networks.kind}}' "${reg_name}")" = 'null' ]; then
  docker network connect "kind" "${reg_name}"
fi

# Document the local registry
# https://github.com/kubernetes/enhancements/tree/master/keps/sig-cluster-lifecycle/generic/1755-communicating-a-local-registry
cat <<EOF | kubectl apply -f -
apiVersion: v1
kind: ConfigMap
metadata:
  name: local-registry-hosting
  namespace: kube-public
data:
  localRegistryHosting.v1: |
    host: "localhost:${reg_port}"
    help: "https://kind.sigs.k8s.io/docs/user/local-registry/"
EOF
```

</details>

I pushed the image to the local registry:
```
docker push localhost:5000/airflow-1.10.6:1.10.15-2
```
Next, I would like to verify that I have a proper image in the local registry:
```
>>> curl http://127.0.0.1:5000/v2/_catalog
{"repositories":["airflow-1.10.6","test-hello-world","test_airflow_with_expose"]}
>>> curl http://127.0.0.1:5000/v2/airflow-1.10.6/tags/list
{"name":"airflow-1.10.6","tags":["1.0","2.0","1.10.15-2","1.0.0","1.10.14-3","1.1.0"]}
```
But if I run `curl http://localhost:5000/v2/_catalog` I would get nothing.

The reason I'm asking is before I am running `helm` with:
```
image:
  repository: localhost:5000/airflow-1.10.6
  version: 1.10.15-2
  pullPolicy: IfNotPresent
```

In the past it worked fine, I was able to curl localhost, but for some time it does not work. What should I do in order to be able to call the local registry via `localhost`, not the `127.0.0.1`?
In the past, when I opened this issue https://github.com/kubernetes-sigs/kind/issues/2600 I didn't have such a problem. 

Versions:
<details>

```
>>> docker version
Client:
 Cloud integration: v1.0.22
 Version:           20.10.12
 API version:       1.41
 Go version:        go1.16.12
 Git commit:        e91ed57
 Built:             Mon Dec 13 11:46:56 2021
 OS/Arch:           darwin/amd64
 Context:           default
 Experimental:      true

Server: Docker Desktop 4.5.0 (74594)
 Engine:
  Version:          20.10.12
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.16.12
  Git commit:       459d0df
  Built:            Mon Dec 13 11:43:56 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.4.12
  GitCommit:        7b11cfaabd73bb80907dd23182b9347b4245eb5d
 runc:
  Version:          1.0.2
  GitCommit:        v1.0.2-0-g52b36a2
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0

>>> kind --version
kind v0.11.1 go1.16.4 darwin/amd64
```

<details>

If there anything I miss, please let me know

**Comment by BenTheElder** [comment]
the current version of the registry script uses port 5001. that may be your issue, given the version you posted in details is also using 5001

**Comment by iamtodor** [comment]
@BenTheElder it worked fine in the past with 5000 port. Let me check 5001 port

**Comment by BenTheElder** [comment]
I mean we changed the port in the script and the version of the script you shared matches that. We had to change the script because 5000 conflicts with a system process on macOS now.

**Comment by iamtodor** [comment]
@BenTheElder yeah, it works now, thanks!
```
>>> curl http://127.0.0.1:5001/v2/_catalog
{"repositories":["airflow-1.10.6"]}
>>> curl http://localhost:5001/v2/_catalog
{"repositories":["airflow-1.10.6"]}
>>> curl http://127.0.0.1:5001/v2/airflow-1.10.6/tags/list
{"name":"airflow-1.10.6","tags":["1.10.15-2"]}
>>> curl http://localhost:5001/v2/airflow-1.10.6/tags/list
{"name":"airflow-1.10.6","tags":["1.10.15-2"]}
```
Could you please point me where to dig if the infra is stuck at init `PodInitializing` status? I am a bit lost. Seems like pulling is fine, but the next stages are not going to be run.

<details>

<summary>kubectl describe deployment airflow</summary>

```
>>> kubectl describe deployment airflow
Name:                   airflow
Namespace:              default
CreationTimestamp:      Mon, 07 Mar 2022 21:06:49 +0100
Labels:                 app=airflow
                        app.kubernetes.io/managed-by=Helm
                        chart=airflow-0.1.1
                        heritage=Helm
                        release=airflow
Annotations:            deployment.kubernetes.io/revision: 1
                        meta.helm.sh/release-name: airflow
                        meta.helm.sh/release-namespace: default
Selector:               app=airflow,app.kubernetes.io/instance=airflow,app.kubernetes.io/name=airflow
Replicas:               1 desired | 1 updated | 1 total | 0 available | 1 unavailable
StrategyType:           RollingUpdate
MinReadySeconds:        0
RollingUpdateStrategy:  25% max unavailable, 25% max surge
Pod Template:
  Labels:  app=airflow
           app.kubernetes.io/instance=airflow
           app.kubernetes.io/name=airflow
  Init Containers:
   init-db:
    Image:      localhost:5001/airflow-1.10.6:1.10.15-2
    Port:       <none>
    Host Port:  <none>
    Args:
      initdb
    Environment:  <none>
    Mounts:
      /opt/airflow/airflow.cfg from airflow-configmap (rw,path="airflow.cfg")
      /opt/airflow/dags from airflow-dag (rw)
  Containers:
   airflow-webserver:
    Image:      localhost:5001/airflow-1.10.6:1.10.15-2
    Port:       8080/TCP
    Host Port:  0/TCP
    Args:
      webserver
    Environment:  <none>
    Mounts:
      /opt/airflow/airflow.cfg from airflow-configmap (rw,path="airflow.cfg")
      /opt/airflow/dags from airflow-dag (rw)
   airflow-scheduler:
    Image:      localhost:5001/airflow-1.10.6:1.10.15-2
    Port:       <none>
    Host Port:  <none>
    Args:
      scheduler
    Environment:  <none>
    Mounts:
      /opt/airflow/airflow.cfg from airflow-configmap (rw,path="airflow.cfg")
      /opt/airflow/dags from airflow-dag (rw)
  Volumes:
   airflow-dag:
    Type:          HostPath (bare host directory volume)
    Path:          /data_dags
    HostPathType:  Directory
   airflow-configmap:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      airflow-configmap
    Optional:  false
   airflow-secret:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  airflow-secret
    Optional:    false
Conditions:
  Type           Status  Reason
  ----           ------  ------
  Available      False   MinimumReplicasUnavailable
  Progressing    False   ProgressDeadlineExceeded
OldReplicaSets:  airflow-8bb784fd4 (1/1 replicas created)
NewReplicaSet:   <none>
Events:
  Type    Reason             Age   From                   Message
  ----    ------             ----  ----                   -------
  Normal  ScalingReplicaSet  14m   deployment-controller  Scaled up replica set airflow-8bb784fd4 to 1
```

</details>

<details>

<summary>kubectl describe pod airflow-8bb784fd4-ncxnd</summary>

```
>>> kubectl describe pod airflow-8bb784fd4-ncxnd
Name:         airflow-8bb784fd4-ncxnd
Namespace:    default
Priority:     0
Node:         kind-control-plane/172.18.0.2
Start Time:   Mon, 07 Mar 2022 21:06:49 +0100
Labels:       app=airflow
              app.kubernetes.io/instance=airflow
              app.kubernetes.io/name=airflow
              pod-template-hash=8bb784fd4
Annotations:  <none>
Status:       Pending
IP:           10.244.0.5
IPs:
  IP:           10.244.0.5
Controlled By:  ReplicaSet/airflow-8bb784fd4
Init Containers:
  init-db:
    Container ID:  containerd://4dfde9f209105c1df577e834cd0a69cbbb80cba670a24e3917201596e1e2d732
    Image:         localhost:5001/airflow-1.10.6:1.10.15-2
    Image ID:      localhost:5001/airflow-1.10.6@sha256:3c6ae8202c8967d2ea61ab4819fa5c84596cb0ee9e7017602b1488d6c8547299
    Port:          <none>
    Host Port:     <none>
    Args:
      initdb
    State:          Running
      Started:      Mon, 07 Mar 2022 21:07:59 +0100
    Last State:     Terminated
      Reason:       Error
      Exit Code:    1
      Started:      Mon, 07 Mar 2022 21:07:41 +0100
      Finished:     Mon, 07 Mar 2022 21:07:45 +0100
    Ready:          False
    Restart Count:  2
    Environment:    <none>
    Mounts:
      /opt/airflow/airflow.cfg from airflow-configmap (rw,path="airflow.cfg")
      /opt/airflow/dags from airflow-dag (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-tfrkh (ro)
Containers:
  airflow-webserver:
    Container ID:
    Image:         localhost:5001/airflow-1.10.6:1.10.15-2
    Image ID:
    Port:          8080/TCP
    Host Port:     0/TCP
    Args:
      webserver
    State:          Waiting
      Reason:       PodInitializing
    Ready:          False
    Restart Count:  0
    Environment:    <none>
    Mounts:
      /opt/airflow/airflow.cfg from airflow-configmap (rw,path="airflow.cfg")
      /opt/airflow/dags from airflow-dag (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-tfrkh (ro)
  airflow-scheduler:
    Container ID:
    Image:         localhost:5001/airflow-1.10.6:1.10.15-2
    Image ID:
    Port:          <none>
    Host Port:     <none>
    Args:
      scheduler
    State:          Waiting
      Reason:       PodInitializing
    Ready:          False
    Restart Count:  0
    Environment:    <none>
    Mounts:
      /opt/airflow/airflow.cfg from airflow-configmap (rw,path="airflow.cfg")
      /opt/airflow/dags from airflow-dag (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-tfrkh (ro)
Conditions:
  Type              Status
  Initialized       False
  Ready             False
  ContainersReady   False
  PodScheduled      True
Volumes:
  airflow-dag:
    Type:          HostPath (bare host directory volume)
    Path:          /data_dags
    HostPathType:  Directory
  airflow-configmap:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      airflow-configmap
    Optional:  false
  airflow-secret:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  airflow-secret
    Optional:    false
  default-token-tfrkh:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  default-token-tfrkh
    Optional:    false
QoS Class:       BestEffort
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                 node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:
  Type     Reason     Age                From               Message
  ----     ------     ----               ----               -------
  Normal   Scheduled  16m                default-scheduler  Successfully assigned default/airflow-8bb784fd4-ncxnd to kind-control-plane
  Normal   Pulling    16m                kubelet            Pulling image "localhost:5001/airflow-1.10.6:1.10.15-2"
  Normal   Pulled     15m                kubelet            Successfully pulled image "localhost:5001/airflow-1.10.6:1.10.15-2" in 45.436845s
  Warning  BackOff    15m                kubelet            Back-off restarting failed container
  Normal   Created    15m (x3 over 15m)  kubelet            Created container init-db
  Normal   Started    15m (x3 over 15m)  kubelet            Started container init-db
  Normal   Pulled     15m (x2 over 15m)  kubelet            Container image "localhost:5001/airflow-1.10.6:1.10.15-2" already present on machine
```

</details>

```
>>> kubectl logs airflow-8bb784fd4-ncxnd airflow-webserver
Error from server (BadRequest): container "airflow-webserver" in pod "airflow-8bb784fd4-ncxnd" is waiting to start: PodInitializing
 >>> kubectl logs airflow-8bb784fd4-ncxnd airflow-scheduler
Error from server (BadRequest): container "airflow-scheduler" in pod "airflow-8bb784fd4-ncxnd" is waiting to start: PodInitializing
```

**Comment by BenTheElder** [comment]
That's a bit out of scope, but you can see in the pod describe output that the init container failed with an error, so that is why the pod is stuck in PodInitializing. You should probably check the logs for the init container.

**Comment by iamtodor** [comment]
@BenTheElder firstly I thought there are only two containers: `airflow-webserver` and `airflow-scheduler`. After that I googled about init container and then I took a look closely: 
`error: a container name must be specified for pod airflow-8bb784fd4-fpg6l, choose one of: [airflow-webserver airflow-scheduler] or one of the init containers: [init-db]`
I think I have realized the issue.

@BenTheElder thank you so much for your time, for your attention to the issue that is not related to kind itself. 
I cannot express how much I'm thankful to you and how much I appreciate your help!
